vault/vault.go

```diff
@@ -551,6 +535,11 @@ func (v *Vault) Undelete(path string) error {
 		version = uint64(respVersions[len(respVersions)-1].Version)
 	}
 
+	err := v.verifySecretUndestroyed(secret, version)
```

Comment:
now that I look at this with full attention, this whole function can be removed now and replaced with
`if respVersions[version].Destroyed { /* err  */ }`. respVersions above would need to be pulled out into the whole function scope, but we already need to get all the versions anyway for this function, so... yeah. Sorry for the runaround.

Reply:
I inlined the verifySecretUndestroyed call as the logic is required if undeleting a version that is not the latest. It removes an additional api call to getting all the versions as well.